api/envoy/api/v2/route/route_components.proto

```diff
@@ -593,6 +593,9 @@ message RouteAction {
     // number is <= the value of the numerator N, or if the key is not present, the default
     // value, the request will be mirrored.
     core.RuntimeFractionalPercent runtime_fraction = 3;
+
+    // Determines if the trace span should be sampled. Defaults to true.
```

Comment:
Can you include links to more detailed explanation of what it means to be sampled vs. non-sampled?

Reply:
I have some more fundamental questions/comments:
1) I think we should call this field something like `enable_tracing` or potentially `trace_sampled` (see next comment)
2) The concept of trace sampling within Envoy is pretty confusing as it's split between Envoy itself and the trace driver. What does this bool actually do? How does it relate to Envoy's trace decision? I assume if Envoy didn't decide to trace the underlying span is a null-span so its basically a NOP? cc @objectiser @rnburn @cgilmour

Reply:
If the request is not traced, then this flag doesn't do anything. The use case is if setting x-b3-sampled to false starting from the mirrored request (some of our users would like to drop sub-traces starting from the mirrored request).

Reply:
OK that's fine. I still think we should make the config flag more clear that it has to do with tracing. It's not clear what it means right now IMO at the shadow level.

Reply:
Renamed to `tracing_sampled`.

Reply:
> I assume if Envoy didn't decide to trace the underlying span is a null-span so its basically a NOP?

Usually the practice is to have un-sampled spans still propagate their context (along with a sampling bit) so that other instrumented services interacting with a request know not to sample it -- so they're not completely a no-op.

Reply:
@rnburn I believe that's what happens after this PR: the mirrored request gets the trace context with a child span of the original request span, along with the sampling bit.